### PR TITLE
Adding MaterialFormattedTextField

### PR DESCRIPTION
**Pull request by Pliavi**

### CODE REVIEW

**Comment by leMaik** [comment]
Thanks for the PR! Couldn't this be done by inheriting from `MaterialTextField`?

**Comment by Pliavi** [comment]
Maybe, the class made to the project was a fast adaptation. I'll try to see what makes the JFormattedTextField formattable (possibly an interface or abstract class) and add only the parts of it to use this inheritance.

**Comment by leMaik** [comment]
@Pliavi That would be awesome!

File: src/main/java/de/craften/ui/swingmaterial/MaterialFormattedTextField.java [hunk 1]
```diff
@@ -63,8 +63,8 @@ public MaterialFormattedTextField(AbstractFormatter formatter) {
 	 * @param factory AbstractFormatterFactory used for formatting.
 	 */
 	public MaterialFormattedTextField(AbstractFormatterFactory factory) {
-		super(factory);
-		initMaterialFormattedTextField();
+		this();
+		setFormatterFactory(factory);
```
Comment (Pliavi) [inline src/main/java/de/craften/ui/swingmaterial/MaterialFormattedTextField.java:75]: The setFormatterFactory()>setValue()>setFormatter() method uses setText() method, that call the FloatingLabel and Line update, but setFormatterFactory() was in the constructor, causing a nullPointerException.

**Comment by leMaik** [comment]
Nice refactoring! :+1:
Please link the material specs (TextField) in the javadoc and feel free to mention yourself as `@author`. Is this ready to be merged?

**Comment by Pliavi** [comment]
OK, i'll add when coming home.
It's not ready yet, this morning I've tested with default value constructor (the JFormattedTextField creates the MaskFormatter) and got another exception, possibly all the constructor will have to be rewrote like those with MaskFormatter in the signature.
But it can be corrected with 2 or 3 lines, today or tomorrow morning it'll be ready to merge.

**Comment by Pliavi** [comment]
Changes made, can be merged now

**Pull request merged**